liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -434,6 +475,51 @@ public String getEEVersion(MavenProject project) {
         return null;
     }
 
+    /**
+     * Returns the MicroProfile major version detected for the given MavenProjects
+     * 
+     * @param mavenProjects project modules, null if a single module project
+     * @return the latest version of MP detected across multiple project modules,
+     *         null if an MP version is not found
+     */
+    public String getMPVersion(List<MavenProject> mavenProjects) {
+        String mpVersion = null;
+        Set<String> mpVersionsDetected = new HashSet<String>();
+        for (MavenProject mavenProject : mavenProjects) {
+            String ver = getMPVersion(mavenProject);
+            if (ver != null) {
+                mpVersionsDetected.add(ver);
+            }
+        }
+        if (!mpVersionsDetected.isEmpty()) {
+            mpVersion = mpVersionsDetected.iterator().next();
+            // if multiple MP versions are found across multiple modules, return the latest
+            // version
+            for (String ver : mpVersionsDetected) {
+                log.warn(ver);
```

Comment:
Did you mean to include this statement? If so could you add some text like "Detected MicroProfile version "+ver ? Or should it be log.debug()?

Reply:
Good catch, I didn't mean to log this. Fixed